survey/survey_views.py

```diff
@@ -244,7 +244,7 @@ def response(request, survey_pk, page_num):
         #these_answer_keys = [k for k in request.POST.keys() if str(q.id) == k[:k.index('_')]]
         if q.required and not these_answers:# (set([str(a.id) for a in q.answer_set.all()]).intersection(set([k for k in request.POST.keys() if request.POST.get(k)!='']))): #or request.POST.getlist(str(q.id))==['']):
             highlight_question.append(q.id)
-            if survey.auto_number: errors.append('Question %d requires a response.'%q_log_list[i][2])
+            if survey.auto_number: errors.append('Question %d requires a response.'%q_log_list[i][0].qnumber)
```

Comment:
Tested it on all answer types and confirmed that it errors with q_log_list[i][2] but works with q_log_list[i][0].qnumber.

While testing though, found out that using answer type "Text" breaks with this error in the view, which I'll open as a separate issue:

    answer.atype not recognized: Tx